FileMapper/fileUploadForm_script.html

```diff
@@ -0,0 +1,232 @@
+      return;
+    }
+
+    // If the file input has an incompatible file
+    if (document.getElementById('extensionError').style.color === 'red') {
+      showMessage(
+          "<strong> Incompatible File Type </strong> <br> Please provide a file compatible to the local path below.");
+      return;
+    }
+
+    document.getElementById('submitButton').value = "Uploading...";
+
+    var file = formObject.uploadedFile.files[0];
+    var fr = new FileReader();
+    fr.fileName = file.name;
+
+    // Event handler executed when the file load event is fired
+    fr.onload = function(e) {
+      var data = e.target.result.replace(/^.*,/, '');
+      var mimetype = e.target.result.match(/^.*(?=;)/)[0];
+      var filename = e.target.fileName;
+
+      // Call the appscript function to create a file in the drive
+      // and on success of that function executes the updateUrl function
+      google.script.run
+          .withSuccessHandler(function(url) {
+            document.getElementById('submitButton').value = "Submit";
+            showMessage(
+                "<strong>Successfully</strong> uploaded the file <strong> " +
+                filename +
+                "</strong> to Google Drive and mapped it to the local path <strong>" +
+                localPathValue +
+                "</strong>. <br> To check it out visit this <strong><a href='" +
+                url + "' style='color: inherit;'>Drive Link</a></strong> !!");
+            setTimeout(function() { google.script.host.close(); }, 2000);
+          })
+          .withFailureHandler(function() {
+            document.getElementById('submitButton').value = "Submit";
+            showMessage(
+                "Failed to upload the file to Google Drive and map it to the local path <strong>" +
+                localPathValue + "</strong>.<br> Please try again.");
+          })
+          .processForm(data, mimetype, filename, localPathValue,
+                      document.getElementById('hiddenId').innerHTML);
+    } 
+    fr.readAsDataURL(file);
+  }
+
+  /**
+   * Get the extension from the file path
+   *
+   * @param {String} path The local path
+   * @return {String} extension The extension of the local path
+   */
+  function getExtensionFromLocalPath(path) {
+    var index = path.lastIndexOf(".");
+    var extension = "";
+    if (index != -1) {
+      extension = path.substr(index + 1);
+    }
+
+    // To emulate Windows behavior
+    // In windows, files without extension are text files
+    return (extension === "") ? "txt" : extension;
+  }
+
+  /**
+   * Function to return the MimeType based on the file extension
+   *
+   * @param {String} extension The extension of the local path
+   * @return {String} MimeType of the file based on it's extension
+   */
+  function getMimetypeFromExtension(extension) {
+    var sheets_extensions = ["xls", "xlsx", "csv", "xlsm", "tsv"];
```

Comment:
I guess this change was done in another PR? once that is merged, sync and re-open this PR

Reply:
This is not the same file, this is on the client-side, that was on the server-side. We can make a call to that same function here but then it will be a call to the server which might take some extra time. What do you suggest?

Reply:
We can have it as a separate js file which could be included both on the server and the client?

Reply:
I couldn't find a way to include a js file on the client-side code in appscript. 
The javascript code can be used from a separate html file using templating but there is no way to use it from a javascript file.